R/zchunk_LA116.geo.R

```diff
@@ -29,57 +33,100 @@ module_energy_LA116.geo_DISABLED <- function(command, ...) {
 
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
-    Land_type_area_ha <- get_data(all_data, "Land_type_area_ha")
+    Land_type_area_ha <- get_data(all_data, "aglu/LDS/Land_type_area_ha")
     A16.geo_curves <- get_data(all_data, "energy/A16.geo_curves")
     A16.EGS_curves <- get_data(all_data, "energy/A16.EGS_curves")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # The method below assumes that the geothermal supply curves (hydrothermal and EGS) have
+    # only one price point per grade (i.e., "grade 2" has the same price in all regions). This
+    # is the standard in GCAM as long as the model has existed, but still this needs to be
+    # confirmed in this initial check
 
-    # Produce outputs
-    # Temporary code below sends back empty data frames marked "don't test"
-    # Note that all precursor names (in `add_precursor`) must be in this chunk's inputs
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    if( nrow(unique(A16.geo_curves[ c("grade", "extractioncost")])) > length( unique( A16.geo_curves$grade ) ) ){
+      stop( "The geothermal (hydrothermal) supply curves have regionally differentiated price points" )
+    }
+    if( nrow(unique(A16.EGS_curves[ c("grade", "extractioncost")])) > length( unique( A16.EGS_curves$grade ) ) ){
```

Comment:
FYI there's also `dplyr::n_distinct` which is faster and clearer than `length(unique(`